### PR TITLE
Adding profile pages

### LINKED ISSUE #69: [US-16] As a user I want to have a User Profile page so that I can see my posts and also my "Followers" and who I "Follow" [9]
As a user, I shall be able to see what I have posted before and be able to see a list of whom I am following and who is following me.  

### **Functional Requirements**

- [x] Another page for the profile 
- [x] The profile page should only include the post that the user has posted 
- [x] The profile page should display a list of who is following and who is following the user. 

### **Specifications**

- [x] Retrieve user post in the database within 2 seconds when refreshing the page 
- [x] Retrieve who is following the user from the database within 3 seconds when refreshing the page
- [x] Retrieve Who the user is following from the database within 3 seconds when refreshing the page 

### **Acceptance Criteria**

- [x] Clicking on your username will redirect you to your profile page
- [x] Profile page will pull all posts made by the user from the database and display it on the webpage
- [x] Have a separate webpage to display information regarding "Followers"
- [x] This webpage should be split into 2 halves, one displaying "Followers" and the other showing "Following"
- [x] "Followers" would display all other users that are following the logged in user 
- [x] "Following" would display all the users that the logged in user is following.

### PR DESCRIPTION
**Pull request by maherysm**
Attempting to work around the conflicts in previous pull request #82 by 

- creating a new branch rebased on [Maxime's branch](https://github.com/halloweentown/SOEN341-W21-A7/tree/userpage)
- then updating that new branch with the code in [Mahery's branch](https://github.com/halloweentown/SOEN341-W21-A7/tree/backend-MahHairy-Profile)

From #82 :
#69 
- Each user has a profile page
- Profiles can be visited via the home page by clicking on the posts usernames
- Profile pages show
  - user details (name, number of followings and followers)
  - user posts
  - follow button (unless the profile belongs to the authenticated user)
  - list of followings and followers when clicking respectively on the Followings and Followers

### CODE REVIEW

**Review by giroux2000 — APPROVED** [review]
Reviewed with Mahery and it worked well

File: idk/app/Models/Following.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
```
Comment (halloweentown) [inline idk/app/Models/Following.php:7]: Comments required

File: idk/app/Models/User.php [hunk 2]
```diff
@@ -38,18 +43,59 @@ class User extends Authenticatable
      *
      * @var array
      */
+    //This part of the code is not used in our implementation, was provided by Laravel.
```
Comment (halloweentown) [inline idk/app/Models/User.php:46]: Can it be commented out then?

File: idk/resources/views/home.blade.php [hunk 3]
```diff
@@ -9,12 +9,14 @@
...
 
                 <div class="card-body">
+                    <!-- This form is filled when the user wants to post a picture. The form takes in a caption and an image.-->
                     <form  action="{{route('home')}}" method="POST" enctype="multipart/form-data" autocomplete="off">
                         {{ csrf_field() }}
                         <textarea name="body" id="caption" rows="3" cols="25" class="form-control" placeholder="Enter your caption here,upload your picture, then press submit!"></textarea>
                         <input type="file" name="image"/>
                         <input type="submit" name="upload" value="Upload"/>
 
+                        <!-- This script limits the amount of words that the user can put in their caption. -->
```
Comment (halloweentown) [inline idk/resources/views/home.blade.php:19]: @MahHairy @giroux2000 is character limit hard, or size based?

@parker-1212 @r-vaidya 
For #85 , include testing of character limit specified here.

**Review by r-vaidya — APPROVED** [review]
Finishes off Sprint 4

**Review by halloweentown — COMMENTED** [review]
see my comments.

**Pull request merged**